src/allotropy/parsers/thermo_fisher_genesys30/thermo_fisher_genesys30_structure.py

```diff
@@ -0,0 +1,147 @@
+from allotropy.parsers.utils.pandas import df_to_series_data
+from allotropy.parsers.utils.uuids import random_uuid_str
+from allotropy.parsers.utils.values import try_float_or_none
+
+
+def create_data(named_file_contents: NamedFileContents) -> Data:
+    """
+    Structures the named_file_contents according to the mapper file
+    :param named_file_contents: The contents of the file to convert
+    :return: The data according to spectrophotometer data mapper
+    """
+    filename = named_file_contents.original_file_name
+
+    lines = read_to_lines(named_file_contents)
+    reader = LinesReader(lines)
+
+    if filename.endswith(".csv"):
+        metadata_list = [line for line in reader.pop_until("^,,") if line]
+        reader.drop_until_inclusive(",,")
+        delimiter = ","
+    elif filename.endswith(".tsv"):
+        metadata_list = list(reader.pop_until_empty())
+        reader.drop_until_inclusive("")
+        delimiter = "\t"
+    else:
+        message = f"{constants.UNSUPPORTED_FILE_FORMAT_ERROR} {filename}"
+        raise AllotropeConversionError(message)
+
+    rawdata_list = list(reader.pop_until_empty())
+    rawdata_string = io.StringIO("\n".join(rawdata_list))
+    rawdata_dataframe = pd.read_csv(rawdata_string, header=0, delimiter=delimiter)
+
+    metadata_string = io.StringIO("\n".join(metadata_list))
+    metadata_dataframe = pd.read_csv(
+        metadata_string,
+        header=None,
+        delimiter=delimiter,
+        keep_default_na=False,
+        index_col=0,
+    ).T
+    metadata_dataframe.columns = metadata_dataframe.columns.str.strip()
+    rawdata_dataframe.columns = rawdata_dataframe.columns.str.strip()
+
+    metadata_dataframe = metadata_dataframe.head(1)
+
+    metadata = df_to_series_data(metadata_dataframe, "Failed to parser header data")
+
+    experiment_name = metadata.get(str, "Scan")
+
+    experiment_type, measurement_time = get_date_time(experiment_name)
+
+    return Data(
+        metadata=Metadata(
+            file_name=filename,
+            device_type=constants.DEVICE_TYPE,
+            device_identifier=NOT_APPLICABLE,
+            model_number=constants.MODEL_NUMBER,
+            software_name=constants.GENESYS_SOFTWARE,
+            detection_type=metadata.get(str, "Mode"),
+            product_manufacturer=constants.PRODUCT_MANUFACTURER,
+            brand_name=constants.BRAND_NAME,
+            container_type=ContainerType.tube,
+        ),
+        measurement_groups=[
+            MeasurementGroup(
+                measurement_time=measurement_time,
+                experiment_type=experiment_type,
+                measurements=[
+                    Measurement(
+                        type_=MeasurementType.ULTRAVIOLET_ABSORBANCE_SPECTRUM,
+                        identifier=random_uuid_str(),
+                        sample_identifier=NOT_APPLICABLE,
+                        operating_minimum=try_float_or_none(metadata.get(str, "Lower")),
```

Comment:
You can just specify float as the type here!
```suggestion
                        operating_minimum=metadata.get(float, "Lower"),
```

Reply:
Okay, Nate. I will start using `get(type, label)` function from now on. Thanks!